geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/CacheClientNotifierTest.java

```diff
@@ -175,4 +177,40 @@ private InternalCacheEvent createMockInternalCacheEvent(
 
     return internalCacheEvent;
   }
+
+  @Test
+  public void testSingletonHasClientProxiesFalseNoCCN() {
+    assertFalse(CacheClientNotifier.singletonHasClientProxies());
+  }
+
+  @Test
+  public void testSingletonHasClientProxiesFalseNoProxy() {
+    InternalCache internalCache = Fakes.cache();
+
+    CacheClientNotifier ccn =
+        CacheClientNotifier.getInstance(internalCache, mock(CacheServerStats.class), 10,
+            10, mock(ConnectionListener.class), null, true);
+
+    assertFalse(CacheClientNotifier.singletonHasClientProxies());
+    ccn.shutdown(111);
+
+  }
+
+  @Test
```

Comment:
Should you also test that addClientInitProxy causes SingletonHasClientProxiesTrue?

Reply:
Thanx for comments. I will update PR.